components/list/test/list.visual-diff.js

```diff
@@ -17,32 +17,134 @@ describe('d2l-list', function() {
+		].forEach((attribute) => {
+			it(`${attribute.name}=${attribute.value}`, async function() {
+				await setAttribute(page, 'd2l-list', attribute.name, attribute.value);
+				if (attribute.listItemAttribute) {
+					await setAttribute(page, 'd2l-list-item', attribute.name, attribute.value);
+				}
+
+				await visualListScreenshot(page, this.test.fullTitle());
+
+				await toggleAttribute(page, 'd2l-list', attribute.name, false);
+				if (attribute.listItemAttribute) {
+					await toggleAttribute(page, 'd2l-list-item', attribute.name, false);
+				}
 			});
 		});
 	});
 
-	it('list-rtl', async function() {
-		const rect = await visualDiff.getRect(page, '#list-rtl');
-		await visualDiff.screenshotAndCompare(page, this.test.fullTitle(), { clip: rect });
+	describe('List tests for selectable', () => {
+		before(async() => {
+			await toggleAttribute(page, 'd2l-list', 'selectable', true);
+		});
+		[
+			{ name: 'hover-effect', value: false },
+			{ name: 'hover-effect', value: true }
+		].forEach((attribute) => {
+			describe(`Selectable when ${attribute.name}=${attribute.value}`, () => {
+				beforeEach(async() =>  {
+					await toggleAttribute(page, 'd2l-list', attribute.name, attribute.value);
+				});
+				afterEach(async() =>  {
+					await toggleAttribute(page, 'd2l-list', attribute.name, false);
+					await resetMouse(page);
+				});
+				it('Checkbox visable', async function() {
+					await visualListScreenshot(page, this.test.fullTitle());
+				});
+
+				it('Hover', async function() {
+					await hover(page, '#hover');
+					await visualListScreenshot(page, this.test.fullTitle());
+				});
+
+				it('Focus', async function() {
+					await tab(page);
+					await visualListScreenshot(page, this.test.fullTitle());
+				});
+
+				it('Selected', async function() {
+					await page.click('#hover');
+					await resetMouse(page); // take away focus.
+					await visualListScreenshot(page, this.test.fullTitle());
+					await page.click('#hover');
+				});
+			});
+		});
 	});
 
+	const toggleAttribute = async(page, selector, name, value) => {
+		await page.$$eval(`#list ${selector}`, (elements, name, value) => {
+			elements.forEach(element => element.toggleAttribute(name, value));
+		}, name, value);
+	};
+	const setAttribute = async(page, selector, name, value) => {
+		await page.$$eval(`#list ${selector}`, (elements, name, value) => {
+			elements.forEach(element => element.setAttribute(name, value));
+		}, name, value);
+	};
+	const visualListScreenshot = async(page, title) => {
+		const rect = await visualDiff.getRect(page, '#list');
+		await visualDiff.screenshotAndCompare(page, title, { clip: rect });
+	};
+	const hover = (page, selector) => {
+		return new Promise((resolve) => {
+			page.hover(`#list ${selector}`);
+			setTimeout(resolve, 505);
+		});
+	};
+	const tab = (page) => {
+		return new Promise((resolve) => {
+			page.keyboard.press('Tab');
+			setTimeout(resolve, 505);
+		});
+	};
+	const resetMouse = (page) => {
```

Comment:
- [ ] visualDiff has removeFocus that can be used instead.

Reply:
Did remove the resetMouse as it also resets the hover. However, I did replace the resetMouse where it was used to reset the focus.